### PR TITLE
Use rcl_clock_t jump callbacks

### PR DESCRIPTION
**Pull request by sloretz**
Relieves rclcpp::TimeSource responsibility of calling jump callbacks.

(Changed branch name to disconnect PRs)

Requires ros2/rcl#284

### CODE REVIEW

File: rclcpp/src/rclcpp/clock.cpp [hunk 1]
```diff
@@ -115,66 +99,51 @@ Clock::get_clock_type()
...
-  // JumpHandler jump_callback;
-  auto jump_callback =
-    std::make_shared<rclcpp::JumpHandler>(pre_callback, post_callback, threshold);
-  {
-    std::lock_guard<std::mutex> guard(callback_list_mutex_);
-    active_jump_handlers_.push_back(jump_callback);
+  rclcpp::JumpHandler * handler = static_cast<rclcpp::JumpHandler *>(user_data);
+  if (before_jump && handler->pre_callback) {
+    handler->pre_callback();
+  } else if (handler->post_callback) {
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/clock.cpp]: The conditions in the `if` and `else if` look weird. If `before_jump` is `true` but the `handler` only has a post callback it is being called here?

Comment (sloretz) [reply rclcpp/src/rclcpp/clock.cpp]: Added test and fixed in af09c599b61f5e73baa204d78705a87cfed32779

**Comment by sloretz** [comment]
CI testing just rclcpp since I don't expect these changes to impact anything downstream.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5104/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1885/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4231/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5082/)

**Comment by sloretz** [comment]
> CI testing just rclcpp since I don't expect these changes to impact anything downstream.

I made a mistake here. tf2_ros uses time jump callbacks and is now broken. Will fix shortly

**Pull request merged**